dev-proxy/ProxyEngine.cs

```diff
@@ -475,10 +475,17 @@ async Task OnRequestAsync(object sender, SessionEventArgs e)
                 await e.GetRequestBodyAsString();
             }
 
-            using var scope = _logger.BeginScope(e.HttpClient.Request.Method ?? "", e.HttpClient.Request.Url, e.GetHashCode());
+            using var scope = _logger?.BeginScope(e.HttpClient.Request.Method ?? "", e.HttpClient.Request.Url, e.GetHashCode());
 
             e.UserData = e.HttpClient.Request;
-            _logger.LogRequest($"{e.HttpClient.Request.Method} {e.HttpClient.Request.Url}", MessageType.InterceptedRequest, new LoggingContext(e));
+
+            _logger?.LogRequest($"{e.HttpClient.Request.Method} {e.HttpClient.Request.Url}", MessageType.InterceptedRequest, new LoggingContext(e));
+
+            if (_config.ShowTimestamps)

```

Comment:
Rather than filtering it here, let's do it in the formatter, like we do for skip-messages:

https://github.com/microsoft/dev-proxy/blob/ea26b9899c4ad001568e9cb4e770d953c8394f09/dev-proxy/Logging/ProxyConsoleFormatter.cs#L50-L54

That way, the code is simpler and the caller doesn't need to be bothered with the conditional logic. This code then simply becomes what's now on line 486.